certification/internal/shell/has_prohibited_packages.go

```diff
@@ -39,16 +53,29 @@ func (p *HasNoProhibitedPackagesCheck) getDataToValidate(image string) (string,
 	return runReport.Stdout, nil
 }
 
-func (p *HasNoProhibitedPackagesCheck) validate(line string) (bool, error) {
-	scanner := bufio.NewScanner(strings.NewReader(line))
+func (p *HasNoProhibitedPackagesCheck) getDataToValidate(dir string) ([]*rpmdb.PackageInfo, error) {
+	log.Debugf("Mounted directory: %s", dir)
+
+	db, err := rpmdb.Open(filepath.Join(dir, "var", "lib", "rpm", "Packages"))
+	if err != nil {
+		return nil, err
+	}
+	pkgList, err := db.ListPackages()
+	if err != nil {
+		return nil, err
+	}
+	return pkgList, nil
+}
+
+func (p *HasNoProhibitedPackagesCheck) validate(pkgList []*rpmdb.PackageInfo) (bool, error) {
```

Comment:
On the whole, I much prefer this more-direct approach of reading the packages in the container. The only caveat I have is the library hasn't seen much maintenance in some time.

Reply:
It's pretty simple. And last update was Dec. 2020, so not too long ago. I'm changing the interface to be more generic. Only the getDataToValidate will have a dependency on the library.